src/ts/lib/api/command.ts

```diff
@@ -1308,8 +1308,10 @@ export const ObjectOpen = (objectId: string, traceId: string, spaceId: string, c
 
 		// Save last opened object
 		const object = detailStore.get(objectId, objectId, []);
+		const windowId = UtilCommon.getCurrentElectronWindowId();
+
 		if (!object._empty_ && ![ I.ObjectLayout.Dashboard ].includes(object.layout)) {
-			Storage.set('lastOpened', { id: object.id, layout: object.layout, spaceId: object.spaceId });
+			Storage.set('lastOpened', { [windowId]: { id: object.id, layout: object.layout, spaceId: object.spaceId }});
```

Comment:
I leverage the fact that the current implementation of `set` merges objects to have a minimally intrusive change.